### ISSUE TITLE
NPE from Fragment onResume

### ISSUE THREAD

**Issue by DavidEdwards** [googlecast/CastCompanionLibrary-android#301]
`mSelectedMedia` [appears to be null at this point of onResume](https://github.com/googlecast/CastCompanionLibrary-android/blob/master/src/com/google/android/libraries/cast/companionlibrary/cast/player/VideoCastControllerFragment.java#L526) in rare circumstances.

```
if (mCastManager.isRemoteMediaPaused() || mCastManager.isRemoteMediaPlaying()) {
	if (mCastManager.getRemoteMediaInformation() != null && mSelectedMedia
		.getContentId().equals(
			mCastManager.getRemoteMediaInformation().getContentId())) {
		mIsFresh = false;
	}
}
```

Here is the stacktrace:

```
Fatal Exception: java.lang.RuntimeException: Unable to resume activity {com.pkg.name/com.google.android.libraries.cast.companionlibrary.cast.player.VideoCastControllerActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.android.gms.cast.y.a()' on a null object reference
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3349)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3380)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2731)
       at android.app.ActivityThread.access$900(ActivityThread.java:172)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1421)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5835)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.android.gms.cast.y.a()' on a null object reference
       at com.google.android.libraries.cast.companionlibrary.cast.player.VideoCastControllerFragment.onResume(SourceFile:516)
       at android.support.v4.app.Fragment.performResume(SourceFile:2020)
       at android.support.v4.app.FragmentManagerImpl.moveToState(SourceFile:1107)
       at android.support.v4.app.FragmentManagerImpl.moveToState(SourceFile:1252)
       at android.support.v4.app.FragmentManagerImpl.moveToState(SourceFile:1234)
       at android.support.v4.app.FragmentManagerImpl.dispatchResume(SourceFile:2056)
       at android.support.v4.app.FragmentController.dispatchResume(SourceFile:196)
       at android.support.v4.app.FragmentActivity.onResumeFragments(SourceFile:505)
       at android.support.v4.app.FragmentActivity.onPostResume(SourceFile:494)
       at android.support.v7.app.AppCompatActivity.onPostResume(SourceFile:174)
       at android.app.Activity.performResume(Activity.java:6356)
       at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3338)
       at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3380)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2731)
       at android.app.ActivityThread.access$900(ActivityThread.java:172)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1421)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5835)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
```

**Comment by naddaf** [comment]
Can you find out what object is null that is causing the NPE?

**Comment by DavidEdwards** [comment]
Unfortunately not, ProGuard doesn't want to reverse the mapping. It is also incredibly rare, so I cannot reproduce it.

You could run [equivalence checks independant of the object](http://developer.android.com/reference/android/text/TextUtils.html#equals(java.lang.CharSequence, java.lang.CharSequence)), just to be certain.

```
if (mCastManager.isRemoteMediaPaused() || mCastManager.isRemoteMediaPlaying()) {
    if (mCastManager.getRemoteMediaInformation() != null && TextUtils.equals(mSelectedMedia
        .getContentId(), mCastManager.getRemoteMediaInformation().getContentId())) {
        mIsFresh = false;
    }
}
```

**Comment by naddaf** [comment]
Addressed in v2.8.4
